[ANDROID] Initial Build Fails (Err with 'order_independent_hash')

**Describe the bug**

Initial build fails

**To Reproduce**
Steps to reproduce the behavior:

- setup android studio 2023.1.1
- Retrieve SDK (standard installation, as asked by Android Studio)
- Use `SDKManager` to install `cmake` and `ndk 25.1.8937393`
- checkout project and submodules
  - commit db2e95a59b74c098f9c782a6317d03bf1623652f (HEAD -> origin/main, tag: ios-v6.0.0-predb2e95a59b74c098f9c782a6317d03bf1623652f)
- load project into Android Studio
- `Android Studio -> Build -> Make Project`


```sh
> Task :MapboxGLAndroidSDK:buildCMakeDebug[armeabi-v7a][mapbox-gl]-2
C/C++: CMake Warning (dev) in CMakeLists.txt:
C/C++:   No project() command is present.  The top-level CMakeLists.txt file must
C/C++:   contain a literal, direct call to the project() command.  Add a line of
C/C++:   code such as
C/C++:     project(ProjectName)
C/C++:   near the top of the file, but after cmake_minimum_required().
C/C++:   CMake is pretending there is a "project(Project)" command on the first
C/C++:   line.
C/C++: This warning is for project developers.  Use -Wno-dev to suppress it.
C/C++: Target platform does not support HTTP tests or dependencies not found.
C/C++: ninja: Entering directory `P:\maplibre\native\platform\android\MapboxGLAndroidSDK\.cxx\Debug\381z5e6m\armeabi-v7a'
C/C++: P:\_prg\AndroidSdk\ndk\25.1.8937393\toolchains\llvm\prebuilt\windows-x86_64\bin\clang++.exe --target=armv7-none-linux-androideabi21 --sysroot=P:/_prg/AndroidSdk/ndk/25.1.8937393/toolchains/llvm/prebuilt/windows-x86_64/sysroot -DMLN_DRAWABLE_RENDERER=1 -DMLN_LEGACY_RENDERER=0 -DMLN_RENDER_BACKEND_OPENGL=1 -DMLN_USE_UNORDERED_DENSE=1 -DRAPIDJSON_HAS_STDSTRING=1 -IP:/maplibre/native/src -IP:/maplibre/native/include -IP:/maplibre/native/platform/default/include -IP:/maplibre/native/vendor/unordered_dense/include -isystem P:/maplibre/native/vendor/mapbox-base/extras/kdbush.hpp/include -isystem P:/maplibre/native/vendor/mapbox-base/deps/supercluster.hpp/include -isystem P:/maplibre/native/vendor/mapbox-base/deps/shelf-pack-cpp/include -isystem P:/maplibre/native/vendor/mapbox-base/deps/geojson-vt-cpp/include -isystem P:/maplibre/native/vendor/mapbox-base/deps/cheap-ruler-cpp/include -isystem P:/maplibre/native/vendor/boost/include -isystem P:/maplibre/native/vendor/csscolorparser -isystem P:/maplibre/native/vendor/earcut.hpp/include -isystem P:/maplibre/native/vendor/eternal/include -isystem P:/maplibre/native/vendor/parsedate -isystem P:/maplibre/native/vendor/polylabel/include -isystem P:/maplibre/native/vendor/protozero/include -isystem P:/maplibre/native/vendor/unique_resource -isystem P:/maplibre/native/vendor/vector-tile/include -isystem P:/maplibre/native/vendor/wagyu/include -isystem P:/maplibre/native/vendor/mapbox-base/include -isystem P:/maplibre/native/vendor/mapbox-base/extras/expected-lite/include -isystem P:/maplibre/native/vendor/mapbox-base/deps/geojson.hpp/include -isystem P:/maplibre/native/vendor/mapbox-base/deps/geometry.hpp/include -isystem P:/maplibre/native/vendor/mapbox-base/deps/jni.hpp/include -isystem P:/maplibre/native/vendor/mapbox-base/deps/optional -isystem P:/maplibre/native/vendor/mapbox-base/deps/pixelmatch-cpp/include -isystem P:/maplibre/native/vendor/mapbox-base/deps/variant/include -isystem P:/maplibre/native/vendor/mapbox-base/extras/rapidjson/include -isystem P:/maplibre/native/vendor/icu/include -isystem P:/maplibre/native/vendor/sqlite/include -g -DANDROID -fdata-sections -ffunction-sections -funwind-tables -fstack-protector-strong -no-canonical-prefixes -D_FORTIFY_SOURCE=2 -march=armv7-a -mthumb -Wformat -Werror=format-security -fexceptions -Qunused-arguments -fno-limit-debug-info  -fPIC -fvisibility=hidden -fvisibility-inlines-hidden -fno-rtti -Wall -Wshadow -Wextra -Werror -Wno-error=tautological-constant-compare -frtti -std=c++17 -MD -MT ../../../../../armeabi-v7a/CMakeFiles/mbgl-core.dir/src/mbgl/gl/renderer_backend.cpp.o -MF ..\..\..\..\..\armeabi-v7a\CMakeFiles\mbgl-core.dir\src\mbgl\gl\renderer_backend.cpp.o.d -o ../../../../../armeabi-v7a/CMakeFiles/mbgl-core.dir/src/mbgl/gl/renderer_backend.cpp.o -c P:/maplibre/native/src/mbgl/gl/renderer_backend.cpp
C/C++: In file included from P:/maplibre/native/src/mbgl/gl/renderer_backend.cpp:10:
C/C++: P:/maplibre/native/include/mbgl/shaders/gl/shader_group_gl.hpp:30:28: error: no matching function for call to 'order_independent_hash'
C/C++:         const size_t key = util::order_independent_hash(propertiesAsUniforms.begin(), propertiesAsUniforms.end());
C/C++:                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
C/C++: P:/maplibre/native/src/mbgl/util/hash.hpp:32:6: note: candidate template ignored: substitution failure [with TIter = std::__wrap_iter<const unsigned int *>, TKey = unsigned int]: non-type template argument evaluates to 1099511628211, which cannot be narrowed to type 'unsigned int'
C/C++: TKey order_independent_hash(TIter cur, const TIter end) {
C/C++:      ^
C/C++: 1 error generated.

> Task :MapboxGLAndroidSDK:buildCMakeDebug[armeabi-v7a][mapbox-gl]-2 FAILED
```


**Platform information**
 - Windows 10 Pro
 - Android Studio 2023.1.1
 - Versions: versions as specified in project configs, nothing changed

--- a/src/mbgl/util/hash.hpp
+++ b/src/mbgl/util/hash.hpp
@@ -25,15 +25,17 @@ template <typename TIter>
 using TIterVal = std::enable_if_t<std::is_integral<typename TIter::value_type>::value, typename TIter::value_type>;
 } // namespace detail
 
-constexpr size_t factor() {
-    if constexpr (sizeof(size_t) == 8) return 1099511628211;
+template <typename T>
+constexpr T factor() {
+    // Default factor prime value from 64-bit FNV hash.
+    if constexpr (sizeof(T) == 8) return 1099511628211;
+    // for 32-bit hash
     return 16777619;
 }
 
 /// Generate a hash key from a collection of integer values which doesn't depend on their order.
 /// Adapted from https://stackoverflow.com/a/76993810/135138
-/// Default factor prime value from 64-bit FNV hash.
-template <typename TIter, typename TKey = detail::TIterVal<TIter>, size_t factor = factor()>
+template <typename TIter, typename TKey = detail::TIterVal<TIter>, TKey factor = factor<TKey>()>
 TKey order_independent_hash(TIter cur, const TIter end) {
     detail::TIterVal<TIter> sum = 0, product = 1;
     for (; cur != end; ++cur) {

